pydicom/dataset.py

```diff
@@ -735,13 +735,18 @@ def convert_pixel_data(self):
                             handlers_tried)
                 self._pixel_array = None
                 self._pixel_id = None
-                if last_exception:
-                    raise last_exception
-                else:
+                exception_messages = []
+                for x in handler_exceptions:
+                    logger.debug("image handler exception", exc_info=x)
+                    exception_messages.append(str(x))
+                if not handler_exceptions:
                     msg = ("No available image handler could "
                            "decode this transfer syntax {}".format(
                                self.file_meta.TransferSyntaxUID))
                     raise NotImplementedError(msg)
+                else:
+                    raise NotImplementedError("\n".join(exception_messages))
+
             # is this guaranteed to work if memory is re-used??
             self._pixel_id = id(self.PixelData)
 
```

Comment:
I think this belongs to the other PR.

Reply:
Yes it does.  Me and Git....

I am making a clean branch now...